kube-runtime/src/utils/predicate.rs

```diff
@@ -152,13 +182,29 @@ where
 
     fn poll_next(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Option<Self::Item>> {
         let mut me = self.project();
+
+        // Evict expired entries before processing new events
+        let now = Instant::now();
+        let ttl = me.config.ttl;
+        me.cache
+            .retain(|_, entry| now.duration_since(entry.last_seen) < ttl);
```

Comment:
I like having it separated from the main body of the poll match like this, but this is probably a debatable point. I.e. whether we should do a O(n) scan in __every__ poll.

A periodic one __could perhaps__ be better for some use cases, but I also like having the code simple. If it's possible to do a periodic one without making it too complex, then maybe that would be worth it.

I'm not sure about this though. Given that most caches are like 10k objects in the heavy case, **if** people do find this to be problematic, then the config could always be extended in the future to only set a `period` for how often to clear the cache.

Reply:
I agree with keeping it simple for now. We can always add a `period` field to the Config later.